roles/keystone/templates/etc/keystone/keystone.conf

```diff
@@ -14,8 +14,55 @@ public_workers = {{ keystone.public_workers }}
 [sql]
 connection=mysql://keystone:{{ secrets.db_password }}@{{ endpoints.db }}/keystone?charset=utf8
 
+[assignment]
+driver = keystone.assignment.backends.sql.Assignment
+
 [identity]
+{% if openldap.enabled|bool %}
+driver = keystone.identity.backends.ldap.Identity
+{% else %}
 driver = keystone.identity.backends.sql.Identity
+{% endif %}
+
+
+{% if openldap.enabled|bool %}
+
+[ldap]
+url = {{ keystone.ldap.url}}
+user = {{ openldap.openldap_user_dn }}
+password = {{ openldap.openldap_user_password }} 
+suffix = {{ openldap.suffix  }}
+use_dumb_member = True
+allow_subtree_delete = false
+query_scope = one
+user_enabled_default = true
+user_enabled_emulation = False
+user_allow_create = true
+user_allow_update = true
+user_allow_delete = true
+user_tree_dn = {{ openldap.users_dn }}
+user_objectclass = inetOrgPerson
+user_id_attribute = cn
+user_name_attribute = sn
+user_mail_attribute = mail
+user_pass_attribute = userPassword
+user_enabled_attribute = employeeType
+user_default_project_id_attribute = departmentNumber
+user_attribute_ignore = tenants
+user_additional_attribute_mapping = description:description
+
+group_tree_dn = {{ openldap.groups_dn }} 
+group_objectclass = groupOfNames
+group_id_attribute = cn
+group_name_attribute = businessCategory 
+group_member_attribute = member
+group_desc_attribute = description
+group_allow_create = true
+group_allow_update = true
+group_allow_delete = true
+use_tls = false
```

Comment:
TLS enablement will need setting up certificates in openldap, we plan to enable this in a future iteration.